# Programming Assignment - 13 - Python Basics

Q.1 : Write a program that calculates and prints the value according to the given formula:

Q = Square root of [(2 * C * D)/H]

Following are the fixed values of C and H:

C is 50. H is 30.

D is the variable whose values should be input to your program in a comma-separated sequence.

Example

Let us assume the following comma separated input sequence is given to the program:

100,150,180

The output of the program should be:

18,22,24


In [19]:
# A.1 :

comma_input = input("Enter a comma seperated numbers : ")
D = comma_input.split(',')
#print(D)
C = 50
H = 30
for i in range(len(D)) :
    Q = ((2 * C * int(D[i])) / H) ** (0.5)
    print(Q,end = ',')

Enter a comma seperated numbers : 100,150,180
18.257418583505537,22.360679774997898,24.49489742783178,

Q.2:
Write a program which takes 2 digits, X,Y as input and generates a 2-dimensional array. The element value in the i-th row and j-th column of the array should be i*j.

Note: i=0,1.., X-1; j=0,1,¡¬Y-1.

Example

Suppose the following inputs are given to the program:

3,5

Then, the output of the program should be:

[[0, 0, 0, 0, 0], [0, 1, 2, 3, 4], [0, 2, 4, 6, 8]] 


In [24]:
# A.2 :

comma_input = input("Enter rows and columns in this format (row,col) : ")
RC = comma_input.split(',')
l = []
diff = 0
for i in range(int(RC[0])) :
    a = 0
    l1 = []
    for j in range(int(RC[1])) :
        l1.append(a)
        a = a + diff
    diff = diff + 1
    l.append(l1)
print(l)

Enter rows and columns in this format (row,col) : 4,6
[[0, 0, 0, 0, 0, 0], [0, 1, 2, 3, 4, 5], [0, 2, 4, 6, 8, 10], [0, 3, 6, 9, 12, 15]]


Q.3 : Write a program that accepts a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:

without,hello,bag,world

Then, the output should be:

bag,hello,without,world


In [3]:
# A.3 : 
comma_input = input("Enter a set of comma seperated words : ")
Words = comma_input.split(',')
#print(Words)
Words_sort = sorted(Words)
s = ','.join(Words_sort)
print("The alphabetically sorted comma seperated word sequence are :",s)

Enter a set of comma seperated words : without,hello,bag,world
The alphabetically sorted comma seperated word sequence are : bag,hello,without,world


Q.4 : Write a program that accepts a sequence of whitespace separated words as 

input and prints the words after removing all duplicate words and sorting them 

alphanumerically.

Suppose the following input is supplied to the program:

hello world and practice makes perfect and hello world again

Then, the output should be:

again and hello makes perfect practice world


In [12]:
# A.4 :

sentence = input("Enter a sentence : ")
sentence_list = sentence.split(' ')
li2 = []
for i in range(len(sentence_list)) :
    if sentence_list[i] not in li2 :
        li2.append(sentence_list[i])
sentence_list = sorted(li2)
sentence2 = ' '.join(sentence_list)
print(sentence2)

Enter a sentence : hello world and practice makes perfect 123 and hello world again 123 456 667 456 123
123 456 667 again and hello makes perfect practice world


Q.5 : Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3

In [14]:
# A.5 :

sentence = input("Enter a sentence : ")
count = 0
count2 = 0
for i in sentence :
    if i.isalpha() :
        count = count + 1
    elif i.isdigit() :
        count2 = count2 + 1
print('LETTERS :',count)
print('DIGITS :',count2)

Enter a sentence : Hello Burhanuddin Kapadia 777 786 45 ! $
LETTERS : 23
DIGITS : 8


Q.6 : A website requires the users to input username and password to register. Write a program to check the validity of password input by users.

Following are the criteria for checking the password:

1. At least 1 letter between [a-z]

2. At least 1 number between [0-9]

1. At least 1 letter between [A-Z]

3. At least 1 character from [$#@]

4. Minimum length of transaction password: 6

5. Maximum length of transaction password: 12

Your program should accept a sequence of comma separated passwords and will check them according to the above criteria. Passwords that match the criteria are to be printed, each separated by a comma.

Example

If the following passwords are given as input to the program:

ABd1234@1,a F1#,2w3E*,2We3345

Then, the output of the program should be:

ABd1234@1


In [24]:
# A.6 :

password_series = input("Enter a sequence of password seperated by comma : ")
pass_list = password_series.split(',')
count1 = 0
count2 = 0
count3 = 0
li2 = []
for i in pass_list :
    if len(i) >= 6 and len(i) <= 12 :
        if i.isalnum() != True :
            for j in i :
                if j.islower() :
                    count1 = count1 + 1
                elif j.isupper() :
                    count2 = count2 + 1
                elif j.isdigit() :
                    count3 = count3 + 1
            if count1 >= 1 and count2 >= 1 and count3 >= 1 :
                li2.append(i)
print(','.join(li2))

Enter a sequence of password seperated by comma : bur@1,786Burh$,Bu110@,burH11,Buq1,vCt112%$
786Burh$,Bu110@,vCt112%$
